srcCpp/perftest_publisher.cxx

```diff
@@ -1836,10 +1849,14 @@ int perftest_cpp::Publisher()
 
                     // flush anything that was previously sent
                     writer->Flush();
-                    writer->wait_for_acknowledgments(
-                            timeout_wait_for_ack_sec,
-                            timeout_wait_for_ack_nsec);
-                    announcement_reader_listener->announced_subscribers = _NumSubscribers;
+                    if (_IsReliable) {
+                        writer->wait_for_acknowledgments(
+                                timeout_wait_for_ack_sec,
+                                timeout_wait_for_ack_nsec);
+                    } else {
+                        MilliSleep(timeout_wait_for_ack_nsec / 1000000);
+                    }
+                    announcement_reader_listener->announced_subscriber_replies = _NumSubscribers;
```

Comment:
There is something I don't understand. Here we assign announced_subscriber_replies to _NumSubscribers. However, in AnnouncementListener::ProcessMessage we increase that value. I do not understand the logic of increasing versus setting to _NumSubscribers

Reply:
@fercsrti77, @softwarejimenez and I realized about the flaws of the current implementation and decided to modify the announcement reader logic using some code suggested by @softwarejimenez. The Pull request has been updated with the new changes. We can go through the changes tomorrow.